pkg/machinehealthcheck/reconciler.go

```diff
@@ -0,0 +1,109 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	anywhereCluster "github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/controller/clientutil"
+	"github.com/aws/eks-anywhere/pkg/controller/serverside"
+)
+
+// Reconciler allows to reconcile machine health checks.
+type Reconciler struct {
+	client    client.Client
+	defaulter anywhereCluster.MachineHealthCheckDefaulter
+}
+
+// New returns a new Reconciler.
+func New(client client.Client, defaulter anywhereCluster.MachineHealthCheckDefaulter) *Reconciler {
+	return &Reconciler{
+		client:    client,
+		defaulter: defaulter,
+	}
+}
+
+// Reconcile installs machine health checks for a given cluster.
+// nolint:gocyclo
+func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) error {
+	log.Info("Installing machine health checks for workload", "cluster", cluster.Name)
+
+	clusterSpec, err := anywhereCluster.BuildSpec(ctx, clientutil.NewKubeClient(r.client), cluster)
```

Comment:
food for thought: if every reconciler ends up building the spec, we might just want to change some/all of them to just take the spec directly and pass it from the cluster controller.

Reply:
https://github.com/aws/eks-anywhere/issues/6301